engine/src/main/resources/assets/shaders/lightGeometryPass_frag.glsl

```diff
@@ -143,21 +147,19 @@ void main() {
 
 #if defined (FEATURE_LIGHT_POINT)
 
-
-    //ref: https://imdoingitwrong.wordpress.com/2011/01/31/light-attenuation/
-    // TODO: Make cutoff a shader parameter
-    const float cutoff= 0.005;
-    const float radius = 3.0;
     // calculate basic attenuation
-    float denom = lightDist/radius + 1;
+    // realistic attenuation ref: https://imdoingitwrong.wordpress.com/2011/01/31/light-attenuation/
+    float denom = lightDist/lightAttenuationRange + 1;
     float attenuation = 1.0/(denom*denom);
-    //float attenuation = clamp (1.0 - ((lightDist * lightDist) / (lightAttenuationRange * lightAttenuationRange)), 0.0, 1.0);
-    //float attenuation = 1.0 - (pow (lightDist, lightAttenuationFalloff) / lightAttenuationRange);
 
-    attenuation = (attenuation - cutoff)/(1.0-cutoff);
+    // Force the light to gradually come to a complete stop at the attenuation range + falloff distance.
+    float lightDistPastRange = max(lightDist - lightAttenuationRange, 0.0);
+    float falloffTerm = 1.0 - min(lightDistPastRange / lightAttenuationFalloff, 1.0);
+    attenuation *= falloffTerm;
```

Comment:
I guess I would suggest renaming falloff to "cutoff". Falloff and attenuation for me are almost synonyms in this context: how the light intensity attenuates/falls with distance. However, here you are operating a CUToff: beyond a certain distance the light intensity is set to zero. 

Again, this might be something we might want to eventually solve in post-processing: the lighting model might produce dimly-lit distant hills as in the "before" picture and as in reality, but the perception model might take those dimly lit pixels and turn them to black if a nearby light source or surface is bright enough.